ui/views/Authentication/ResetPasswordView.tsx

```diff
@@ -0,0 +1,168 @@
+import { ArrowBack } from '@mui/icons-material';
+import { Button, CircularProgress, TextField } from '@mui/material';
+import classnames from 'classnames';
+import { Formik, FormikErrors, FormikHelpers } from 'formik';
+import { useSnackbar } from 'notistack';
+import React, { Fragment, useEffect } from 'react';
+import { Link as RouterLink, useLocation, useNavigate } from 'react-router-dom';
+import Logo from 'assets';
+import useResetPassword from 'shared/authentication/actions/resetPassword';
+
+interface ResetPasswordValues {
+  password: string;
+  verifyPassword: string;
+}
+
+const initialValues: ResetPasswordValues = {
+  password: '',
+  verifyPassword: '',
+};
+
+export default function ResetPasswordView(): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const resetPassword = useResetPassword();
+
+  const search = location.search;
+  const query = new URLSearchParams(search);
+  const token = query.get('token');
+
+  useEffect(() => {
+    if (!token) {
+      navigate('/login');
+      enqueueSnackbar('You must get a password reset link to change your password.', {
+        variant: 'warning',
+        disableWindowBlurListener: true,
+      });
+    }
+  }, [token, enqueueSnackbar, navigate]);
```

Comment:
If the token is provided, is there an easy way we can clear it from the URL? Makes the url prettier but also prevents the user accidently navigating to it again through the back button after the token has been used.

Reply:
@th3fallen do you know how I could change the URL without re-rendering this boi?